src/components/SSOLanding.js

```diff
@@ -1,82 +1,57 @@
 import _ from 'lodash';
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { withRouter } from 'react-router';
-import { withCookies, Cookies } from 'react-cookie';
+import React, { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
 import queryString from 'query-string';
-import { requestUserWithPerms } from '../loginServices';
+import { useStore } from 'react-redux';
+import { okapi } from 'stripes-config';
 
-const requestUserWithPermsDeb = _.debounce(requestUserWithPerms, 5000, { leading: true, trailing: false });
 
-class SSOLanding extends Component {
-  constructor(props, context) {
-    super(props, context);
+import { requestUserWithPerms } from '../loginServices';
 
-    this.store = context.store;
-    this.sys = require('stripes-config'); // eslint-disable-line global-require
-    this.okapiUrl = this.sys.okapi.url;
-    this.tenant = this.sys.okapi.tenant;
-  }
+const requestUserWithPermsDeb = _.debounce(requestUserWithPerms, 5000, { leading: true, trailing: false });
 
-  componentDidMount() {
-    const token = this.getToken();
-    if (token) {
-      requestUserWithPermsDeb(this.okapiUrl, this.store, this.tenant, token);
-    }
-  }
+const SSOLanding = () => {
+  const location = useLocation();
+  const [cookies] = useCookies(['ssoToken']);
+  const store = useStore();
 
-  getParams() {
-    const search = this.props.location.search;
+  const getParams = () => {
+    const search = location.search;
     if (!search) return undefined;
     return queryString.parse(search) || {};
-  }
-
-  getToken() {
-    const params = this.getParams();
-    const cookies = this.props.cookies;
-    return cookies.get('ssoToken') || params.ssoToken;
-  }
+  };
 
-  render() {
-    const params = this.getParams();
-    const token = this.getToken();
+  const getToken = () => {
+    const params = getParams();
+    return cookies.ssoToken || params.ssoToken;
+  };
 
-    if (!token) {
-      return (
-        <div>
-          No
-          {' '}
-          <tt>ssoToken</tt>
-          {' '}
-          cookie or query parameter
-        </div>
-      );
+  useEffect(() => {
+    const token = getToken();
+    if (token) {
+      requestUserWithPermsDeb(okapi.url, store, okapi.tenant, token);
     }
+  });
```

Comment:
@zburke should we add `[token]` here? and perhaps move `const token = getToken();` out of `useEffect`?

Otherwise I think this will execute on every rerender (which perhaps is intentional here?).